### PR TITLE
Mob refactoring

### PR DESCRIPTION
**Pull request by russgold**
This includes the refactoring work done in the mob session and refactorings originally proposed in owls-83432-large_result_sets

### CODE REVIEW

**Review by rjeberhard — APPROVED** [review]

**Review by lennyphan — APPROVED** [review]

File: operator/src/test/java/oracle/kubernetes/operator/MainTest.java [hunk 1]
```diff
@@ -274,14 +228,14 @@ public void deleteDomainPresenceWithTimeCheck_doNotDelete_with_older_DateTime()
...
-  private Collection<String> invoke_getDomainNamespaces(String tnValue, String namespace)
-      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-    if (getDomainNamespaces == null) {
-      getDomainNamespaces =
-          Main.class.getDeclaredMethod("getDomainNamespacesList", String.class, String.class);
-      getDomainNamespaces.setAccessible(true);
-    }
-    return (Collection<String>) getDomainNamespaces.invoke(null, tnValue, namespace);
+  @Test
+  public void afterReadingExistingResourcesForNamespace_WatcheraAreDefined() {
```
Comment (ankedia) [inline operator/src/test/java/oracle/kubernetes/operator/MainTest.java:232]: It looks like there's a typo for "Watchers" in method name `afterReadingExistingResourcesForNamespace_WatcheraAreDefined`

File: operator/src/main/java/oracle/kubernetes/operator/Main.java [hunk 2]
```diff
@@ -333,38 +327,41 @@ static Step createDomainRecheckSteps(DateTime now) {
...
-        ConfigMapHelper.createScriptConfigMapStep(operatorNamespace, ns),
-        createConfigMapStep(ns),
-        readExistingPods(ns),
-        readExistingEvents(ns),
-        readExistingServices(ns),
-        readExistingDomains(ns));
-  }
+  static Step readExistingResources(String ns) {
+    NamespacedResources resources = new NamespacedResources(ns, null);
+    DomainResourcesValidation dpis = new DomainResourcesValidation(ns, processor);
```
Comment (ankedia) [inline operator/src/main/java/oracle/kubernetes/operator/Main.java:332]: Should the variable name 'dpis' be something like 'drv' so it's not confused with DomainPresenceInfo?

File: operator/src/test/java/oracle/kubernetes/operator/MainTest.java (hunk 1 above)
Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/MainTest.java:232]: You're correct, and I am sorry I didn't notice it sooner.  I'll try to include it in a future PR.

File: operator/src/main/java/oracle/kubernetes/operator/Main.java (hunk 2 above)
Comment (russgold) [reply operator/src/main/java/oracle/kubernetes/operator/Main.java:332]: That's a good idea.

**Pull request merged**